### ISSUE TITLE
[$2000] Back from user details opens group chat instead of group details on android chrome

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#23935]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open the app in android chrome
2. Open any group report
3. Click on header to open details page
4. Click on any user
5. Click on "message %user%"
6. Click back button in app and observe that it open group chat instead of group details page as we do on web chrome
## Expected Result: 
App should open group details page on back from user report if we open user report from group details page
## Actual Result: 
App opens group report instead of group details page on back from user report if we open user report from group details page on android chrome

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [x] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.47-3
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/0f8d9ced-21a3-478e-9a63-4a870d142048

https://github.com/Expensify/App/assets/43996225/a537bfd8-c936-43ed-a357-2c3589e1cbe0

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dhanashree-sawant
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1690437966940519

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01d267fead81e68c17</li>
        <li>Upwork Job ID: 1687163869748379648</li>
        <li>Last Price Increase: 2023-08-15</li>
    </ul>
</details>

**Comment by ygshbht** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The issue at hand involves incorrect navigation behavior in Android Chrome: the application returns to the group chat instead of the group details page after messaging a user from the details page.

### What is the root cause of that problem?
The root cause of this problem lies in the `navigateToAndOpenReportWithAccountIDs `function. When this function is called upon clicking the 'message' button, it dismisses the current Right Hand Navigation (RHN) screen, causing it to be wiped from the navigation history. As a result, when the back button is pressed, the application doesn't navigate back to the group details page (as it's no longer in the history stack), and instead opens the group chat.

### What changes do you think we should make in order to solve the problem?
To fix this issue, we should modify the `navigateToAndOpenReportWithAccountIDs `function to navigate to the new report route instead of dismissing it. This way, the RHN screen (group details page in this case) will be kept in the navigation history and we can navigate back to it when the back button is pressed. Here's the proposed code modification:

```
function navigateToAndOpenReportWithAccountIDs(participantAccountIDs) {
    let newChat = {};
    const chat = ReportUtils.getChatByParticipants(participantAccountIDs);
    if (!chat) {
        newChat = ReportUtils.buildOptimisticChatReport(participantAccountIDs);
    }
    const reportID = chat ? chat.reportID : newChat.reportID;

    // We want to pass newChat here because if anything is passed in that param (even an existing chat), we will try to create a chat on the server
    openReport(reportID, [], newChat, '0', false, participantAccountIDs);
    Navigation.navigate(ROUTES.getReportRoute(reportID));
}
```

<sub>This fix, however, will highlight another issue: the inability to select the Left Hand Navigation (LHN) after the RHN is closed, due to an improperly positioned 'nodropzone' element. This will require a separate fix. The root cause of this issue should be investigated, and the position or behavior of the 'nodropzone' element should be adjusted accordingly to allow proper interaction with the LHN after the RHN is closed.</sub>

**Comment by jeet-dhandha** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
- On back from second report screen the profile page is not opened instead the first chat screen is opened.

### What is the root cause of that problem?
- The modal is dismissed and then we are redirected to the second chat screen thus updating the root state to:
    - Home
    - CentralPanNavigator - (first chat)
    - CentralPanNavigator - (secon chat)

### What changes do you think we should make in order to solve the problem?
1. We should add two params in `navigateToAndOpenReportWithAccountIDs` function

https://github.com/Expensify/App/blob/30fc9bf74f71c2c1f9e72e924c3013a32514a840/src/libs/actions/Report.js#L548C1-L553C72

    - Params:
        - `shouldDismissModal` - to dismiss the modal
        - `pushToNavigationState` - to push the navigation state to the root state

2. Then we update the navigation code as based on above params.
3. We add code to PUSH report to the navigation state if `pushToNavigationState` is true. (Refer below code if needed).

<details>

<summary>Code Changes</summary>

```js
function navigateToAndOpenReportWithAccountIDs(participantAccountIDs, shouldDismissModal = true, pushToNavigationState = false) {
    ...rest code
    if (shouldDismissModal) {
        Navigation.dismissModal(reportID);
    }

    if (pushToNavigationState) {
        Navigation.navigate(ROUTES.getReportRoute(reportID), 'PUSH');
    }
}
```

</details>

4. Now we create a `isSmallDevice` constant which is basically this conditions `Browser.isMobile() || isSmallScreenWidth` and use it to decide whether to push the report to navigation state or not.

5. Finally we update `Report.navigateToAndOpenReportWithAccountIDs([accountID])` to `Report.navigateToAndOpenReportWithAccountIDs([accountID], !isSmallDevice, isSmallDevice)` in `ProfilePage.js` File.

### What alternative solutions did you explore? (Optional)
- N/A

**Comment by ygshbht** [comment]
> <sub>This fix, however, will highlight another issue: the inability to select the Left Hand Navigation (LHN) after the RHN is closed, due to an improperly positioned 'nodropzone' element. This will require a separate fix. The root cause of this issue should be investigated, and the position or behavior of the 'nodropzone' element should be adjusted accordingly to allow proper interaction with the LHN after the RHN is closed.
>https://github.com/Expensify/App/issues/23885
>https://github.com/Expensify/App/issues/23881

The issue mentioned at the bottom might be fixed by this [PR](https://github.com/Expensify/App/pull/23856)

**Comment by michaelhaxhiu** [comment]
Hm I can't reproduce this one on my android (pixel / android / google chrome / staging).

**Comment by michaelhaxhiu** [comment]
https://github.com/Expensify/App/assets/13206896/0a291e4e-03a5-40d5-9a09-e3d128c7a124

Here's a recording. Am I missing a step here @dhanashree-sawant?

**Comment by michaelhaxhiu** [comment]
I'm leaning toward closing this one as-is until we get more clarity around the steps to take.

**Comment by ygshbht** [comment]
@michaelhaxhiu  Yeah, you forgot to click the "Message user" button

**Comment by dhanashree-sawant** [comment]
Hi @michaelhaxhiu, yes you forgot to click on 'Message user' after opening user details

**Comment by michaelhaxhiu** [comment]
Note: I'm preparing to go OOO for ~2 weeks and need a BZ buddy to watch over this in the meantime. 🙏

Next step: Please ensure that we push this forward and double price if no winning proposal in the next week (by 8/11).

Thanks in advance @jliexpensify, I can take this back when I get back if it's not complete before then.

**Comment by jliexpensify** [comment]
Not overdue!

**Comment by jliexpensify** [comment]
Not overdue, waiting on proposals

**Comment by jliexpensify** [comment]
No proposals yet, going to double this.

**Comment by mountiny** [comment]
> @mountiny can you please confirm that it's intentional to close all RHP modals (on mobile) when open DM chat by clicking "Message xxx" on profile page?

I think this is fine and intentional to make sure that the LHN is closer in the stack

I am a little bit confused by this issue and honestly it feels so minor we could close this out. The mWeb / native discrepancies are hard

**Comment by jliexpensify** [comment]
@himanshuragi456 Hi there - it's the C+ that will consider proposals, I'm just here to push the issue along and issue payments!

Ok cool, going to close this one then - based of Vit's comment:

> I am a little bit confused by this issue and honestly it feels so minor we could close this out. The mWeb / native discrepancies are hard
